src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,245 @@
+            }
+        }
+    }
+    
+    TEST_CASE(WriteJPG_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.jpg";
+        
+        {
+            Image image(2, 2, 2, 2, 3, PixelFormatFloat);
+            
+            image.clear(Color3f(0.6f, 0.7f, 0.8f));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color3f c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_FEQ_EPS(Color3f(0.6f, 0.7f, 0.8f), c, 0.003f);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WritePNG_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.png";
+        
+        {
+            Image image(2, 2, 2, 2, 4, PixelFormatFloat);
+            Color4b color(50, 100, 150, 42);
+            color.unpremultiply_in_place();
+            color.rgb() = fast_linear_rgb_to_srgb(color.rgb());
+            color = saturate(color);
+            color.premultiply_in_place();
+            
+            image.clear(color);
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+            
+            Color4b color(50, 100, 150, 42);
+            color.unpremultiply_in_place();
+            color.rgb() = fast_linear_rgb_to_srgb(color.rgb());
+            color = saturate(color);
+            color.premultiply_in_place();
+            
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color4b c;
+                    image->get_pixel(x, y, c);
+                    
+                    EXPECT_EQ(color, c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteTIFF_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.tiff";
+        
+        {
+            Image image(16, 16, 16, 16, 4, PixelFormatUInt8);
+            
+            image.clear(Color4b(50, 100, 150, 42));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 16; ++y)
```

Comment:
There is a lot of duplicated code . I'm sure we can create functions for this. @FrancoisBeaune What do you think ? 

The process is always:
1. Create a color 
2. Write an image of given size with this color
3. Read the image and assert that all pixels are == to the color

Reply:
I totally agree there's too much duplicate code. Though I wonder if we should keep the test code explicit.

Reply:
It can stay explicit if you add functions. Take a look at some existing tests, we already do this kind of stuff :)